src/metabase/db/metadata_queries.clj

```diff
@@ -27,12 +27,52 @@
       :data
       :rows))
 
+(defn- partition-field->filter-form
+  "Given a partition field, returns the default value can be used to query."
+  [field]
+  (let [field-form [:field (:id field) nil]]
+    (condp #(isa? %2 %1) (:base_type field)
+      :type/Number   [:> field-form -9223372036854775808]
+      :type/Date     [:> field-form "0001-01-01"]
+      :type/DateTime [:> field-form "0001-01-01 00:00:00"])))
+
+(defn- query-with-default-partitioned-field-filter
+  [query table-id]
+  (let [;; In bigquery, range or datetime partitioned table can have only one partioned field,
+        ;; Ingestion time partitioned table can use either _PARTITIONDATE or _PARTITIONTIME as
+        ;; partitioned field
+        partition-field (or (t2/select-one :model/Field
+                                           :table_id table-id
+                                           :database_partitioned true
+                                           :active true
+                                           ;; prefer _PARTITIONDATE over _PARTITIONTIME for ingestion time query
+                                           {:order-by [[:name :asc]]})
+                            (throw (ex-info (format "No partitioned field found for table: %d" table-id)
+                                            {:table_id table-id})))
+        filter-form     (partition-field->filter-form partition-field)]
+    (update query :filter (fn [existing-filter]
+                            (if (some? existing-filter)
+                              [:and existing-filter filter-form]
+                              filter-form)))))
+
+(defn- field-mbql-query
```

Comment:
this function is used in `field-distinct-values`, which is used to get a list of values for field values during the sync process.